### PR TITLE
Adds feature useOptional

### PR DESCRIPTION
**Pull request by huningd**
### PR checklist

- [X] Read the [contribution guildelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [X] Ran the shell/batch script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates)
- [X] Filed the PR against the correct branch: master for non-breaking changes and `2.3.0` branch for breaking (non-backward compatible) changes.

### Description of the PR

This PR adds the feature useOptional to the JavaSpring generator. For non required parameters this feature will generate Optional types. Through this it is more clearly that a parameter isn't required. The feature useOptional is deactivates by default.

solves #3819

### CODE REVIEW

File: modules/swagger-codegen/src/test/java/io/swagger/codegen/options/SpringOptionsProvider.java [hunk 1]
```diff
@@ -19,6 +19,7 @@
     public static final String RESPONSE_WRAPPER = "Callable";
     public static final String USE_TAGS = "useTags";
     public static final String USE_BEANVALIDATION = "false";
+	public static final String USE_OPTIONAL = "false";
```
Comment (cbornet) [inline modules/swagger-codegen/src/test/java/io/swagger/codegen/options/SpringOptionsProvider.java]: Alignment issue

File: modules/swagger-codegen/src/test/java/io/swagger/codegen/options/SpringOptionsProvider.java [hunk 2]
```diff
@@ -40,6 +41,7 @@ public String getLanguage() {
         options.put(SpringCodegen.RESPONSE_WRAPPER, RESPONSE_WRAPPER);
         options.put(SpringCodegen.USE_TAGS, USE_TAGS);
         options.put(SpringCodegen.USE_BEANVALIDATION, USE_BEANVALIDATION);
+		options.put(SpringCodegen.USE_OPTIONAL, USE_OPTIONAL);
```
Comment (cbornet) [inline modules/swagger-codegen/src/test/java/io/swagger/codegen/options/SpringOptionsProvider.java]: Here also

**Comment by cbornet** [comment]
@huningd the whole commit is badly indented. Can you set your indentation to 4 spaces ?

**Comment by wing328** [comment]
@huningd do you need help fixing the code format?

File: modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/SpringCodegen.java [hunk 3]
```diff
@@ -35,6 +37,7 @@
     protected String responseWrapper = "";
     protected boolean useTags = false;
 	protected boolean useBeanValidation = true;
+	protected boolean useOptional = true;
```
Comment (wing328) [inline modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/SpringCodegen.java]: @huningd I'll change this to `false` as part of the formatting change.

cc @cbornet

**Comment by wing328** [comment]
Closed via https://github.com/swagger-api/swagger-codegen/pull/5968

**Pull request closed without merging**